src/Uno.UI/UI/Xaml/ApplicationActivity.Android.cs

```diff
@@ -96,19 +97,18 @@ public void ExitFullscreen()
 			Window.ClearFlags(WindowManagerFlags.Fullscreen);
 		}
 
-		private void OnLayoutChanged(Rect occludedRect)
+		private void OnLayoutChanged(Rect statusBar, Rect keyboard, Rect navigationBar)
 		{
-			if(_inputPane.OccludedRect != occludedRect)
-			{
-				_inputPane.OccludedRect = ViewHelper.PhysicalToLogicalPixels(occludedRect);
-			}
+			Xaml.Window.Current?.RaiseNativeSizeChanged();
+			_inputPane.OccludedRect = ViewHelper.PhysicalToLogicalPixels(keyboard);
```

Comment:
Shouldn't we check if the sizes are differents (keyboard vs Occluded) ?
In the case of when you just hide / show your navigation bar, it will not change right ? I don't know if it will cost us some performances to check it.

Reply:
`InputPane.OccludedRect::set` has built-in value changed guard. Also, the existing value-changed guard is useless here, since it is checking physical against logical values which are never the same, exception when it is 0-sized, or the device has a display scaling of 1:1.